web_poet/fields.py

```diff
@@ -24,31 +24,48 @@ async def to_item(self):
             return await item_from_fields(self)
 
 """
+from functools import update_wrapper
 from types import MethodType
 
-from web_poet.utils import maybe_await
+from web_poet.utils import cached_method, maybe_await
 
 
-class field:
+def field(method=None, *, cached=False):
     """
-    Page Object methods decorated with ``field`` decorator
+    Page Object methods decorated with ``@field`` decorator
     are called by :func:`item_from_fields` or :func:`item_from_fields_sync`
     to populate item attributes.
-    """
-
-    def __init__(self, meth):
-        if not callable(meth):
-            raise TypeError(f"@field decorator must be used on methods, {meth!r} is decorated instead")
-        self.meth = meth
 
-    def __set_name__(self, owner, name):
-        if not hasattr(owner, "_auto_item_fields"):
-            # dict is used instead of set to preserve the insertion order
-            owner._auto_item_fields = {}
-        owner._auto_item_fields[name] = True
+    By default, decorating a method with ``@field`` decorator make them cache
+    their results. Use ``@field(cached=False)`` to disable this behavior.
```

Comment:
```suggestion
    Use ``@field(cached=True)`` to cache the method result.
```

Reply:
We could also mention that the cache replacement policy is LRU.

Reply:
It's not, the cache is unbounded.